src/ucp/wireup/select.c

```diff
@@ -2363,12 +2367,25 @@ ucp_wireup_select_aux_transport(ucp_ep_h ep, unsigned ep_init_flags,
 {
     ucp_wireup_select_context_t select_ctx = {};
     ucp_wireup_criteria_t criteria         = {};
+    unsigned prefer_cb_async_cap           = 1;
     ucp_wireup_select_params_t select_params;
+    ucs_status_t status;
 
     ucp_wireup_select_params_init(&select_params, ep, ep_init_flags,
                                   remote_address, tl_bitmap, 1);
-    ucp_wireup_fill_aux_criteria(&criteria, ep_init_flags);
-    return ucp_wireup_select_transport(&select_ctx, &select_params, &criteria,
-                                       ucp_tl_bitmap_max, UINT64_MAX,
-                                       UINT64_MAX, UINT64_MAX, 1, select_info);
+    ucp_wireup_fill_aux_criteria(&criteria, ep_init_flags, prefer_cb_async_cap);
+    status = ucp_wireup_select_transport(&select_ctx, &select_params,
+                                         &criteria, ucp_tl_bitmap_max,
+                                         UINT64_MAX, UINT64_MAX, UINT64_MAX, 0,
+                                         select_info);
+    if (status == UCS_OK) {
+        return UCS_OK;
+    }
+
+    prefer_cb_async_cap = 0;
+    ucp_wireup_fill_aux_criteria(&criteria, ep_init_flags, prefer_cb_async_cap);
```

Comment:
Add a comment before /* Fallback to an auxiliary transport without async active message callback requirement */

Reply:
Done